src/components/Datepicker/Datepicker.jsx

```diff
@@ -172,28 +172,47 @@ const DatePicker = ({
     (e) => {
       const idx = parseInt(getDID(e), 10);
       const labelMonth = labelMonthsState[idx];
-      const nextMonth = labelMonth + 1;
-      if (nextMonth <= 12) {
-        const newLabelMonthsState = [...labelMonthsState];
+      let nextMonth = labelMonth + 1;
+      const newLabelMonthsState = [...labelMonthsState];
+      if (labelMonth < 12) {
```

Comment:
you are using the label here, you should use the index instead!!!
the label is for displaying, so you shouldn't use it in your logic.

Reply:
There are no month index on the old code case. Only year index. You can go to the file and see entirely

Reply:
see line 173: `const idx = parseInt(getDID(e), 10);` shouldn't you use that index ?
what if the months labels are `Jan, Feb, ...` ?

Reply:
you can verify that with trying string month labels

Reply:
After dicussion on riot. We decided to keep the logic and just change the variable name. Since the variable name was not expressed what it does correctly